src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1222,24 +1222,47 @@ bool LibraryCallKit::inline_vector_mem_masked_operation(bool is_store) {
 
   int mem_num_elem = using_byte_array ? num_elem * type2aelembytes(elem_bt) : num_elem;
   BasicType mem_elem_bt = using_byte_array ? T_BYTE : elem_bt;
-  bool use_predicate = arch_supports_vector(is_store ? Op_StoreVectorMasked : Op_LoadVectorMasked,
-                                            mem_num_elem, mem_elem_bt,
-                                            (VectorMaskUseType) (VecMaskUseLoad | VecMaskUsePred));
-  // Masked vector store operation needs the architecture predicate feature. We need to check
-  // whether the predicated vector operation is supported by backend.
-  if (is_store && !use_predicate) {
-    if (C->print_intrinsics()) {
-      tty->print_cr("  ** not supported: op=storeMasked vlen=%d etype=%s using_byte_array=%d",
-                    num_elem, type2name(elem_bt), using_byte_array ? 1 : 0);
+  bool supports_predicate = arch_supports_vector(is_store ? Op_StoreVectorMasked : Op_LoadVectorMasked,
+                                                mem_num_elem, mem_elem_bt,
+                                                (VectorMaskUseType) (VecMaskUseLoad | VecMaskUsePred));
+
+  // If current arch does not support the predicated operations, we have to bail
+  // out when current case uses the predicate feature.
+  if (!supports_predicate) {
+    bool use_predicate = false;
```

Comment:
If we rename this to needs_predicate it will be easier to understand.

Reply:
Thanks for the comment! This local variable will be removed after adding the similar intrinsify for store masked. Please help to see the PR https://github.com/openjdk/jdk/pull/8544. Thanks so much!

Reply:
Renamed to "needs_predicate". Thanks!